feat: React login form against the SecureDrop demo server

cfm opened the pull request:
Closes #4.

On this branch, `npm i && npm run dev` will open a single-page application that lets you "log in" to the Journalist API on the SecureDrop demo server.  Really, you're just sending an [authentication request](https://developers.securedrop.org/en/latest/journalist_api.html#authentication) and, if successful, caching some user-level information for display.  But this lets us demonstrate, commit by commit:

1. scaffolding an application in React;
2. mocking out a minimal UI with React Bootstrap;
3. making API calls, albeit without any request or state-machine abstractions yet; and
4. mutating structured state in a disciplined way and having it propagate automatically.

cfm committed:
chore: "npm create vite@latest securedrop-gui-exploration -- --template react-ts"

Plus the initial "npm install".

cfm committed:
style: Prettier

cfm committed:
chore: "npm install --save bootstrap react-bootstrap"

cfm committed:
chore: "npm install --save @reduxjs/toolkit react-redux"

cfm committed:
chore: set up Redux store

cfm committed:
refactor(counter): Redux

cfm committed:
feat: login form; "session" store

See the readme and "sessionSlice.ts" for limitations: this is for
demonstration only!

cfm committed:
feat: authenticate against demo-journalist.securedrop.org (proxied via
Vite)

cfm committed:
feat: prefill and document demo values

cfm committed:
refactor: group inputs

cfm committed:
feat: friendly/verbose greetings

legoktm commented:
Quick thoughts:
* I think we can use sd-proxy in the dev environment - no two different paths between dev and prod :)
* Should login happen in the backend so the authentication token can be kept out of the renderer?
   * Probably, though I think this depends on the whole frontend/backend thing, because if we expose the whole API interface to frontend and then just authenticate the request in the backend, that doesn't really provide much benefit from preventing malicious requests (just prevents against exfiltration of the token)
* We'll have to merge all the vite/ts/etc. stuff that the react template created with the stuff the electron one did - shouldn't be too bad.
* +1 for prettier.

cfm commented:
Oh, I agree for integration with a back-end framework, @legoktm.  But I understood proxying to be out of scope here: all front end, no back end until integrated with one of those frameworks.

cfm committed:
chore: move to framework-specific subdirectory

---since we want to be able to merge parallel implementations, using
different frameworks, from different pull requests.

zenmonkeykstop commented:
Can this be built as a standalone app for comparison purposes?  Commands above opened it as an SPA.

cfm committed:
style: placate tsc "noUnusedLocals"

cfm commented:
@zenmonkeykstop, I'm not seeing options for further bundling *other* than something like Electron or Tauri.  My mental model is that we intended this to be a shared *prerequisite* for testing out those frameworks, as we discussed briefly in stand-up just now, not for direct comparison with them.  But let's chat if I'm missing something.

legoktm commented:
Gonna move the LICENSE file back to the root and then merge this.
